### PR TITLE
Game logic

### PR DESCRIPTION
**Pull request by tnguyen57**
Draft Pull Request

### CODE REVIEW

File: server/GameBoardData.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
+
+/**
+ * This class will store the game state of a players board on the server.
+ * This class will store current guesses and ship positions
+ */
+
+require("GameShipData");
+require("./utilities");
```
Comment (kvverti) [inline server/GameBoardData.js]: These need to be assigned to things.

```
const GameShipData = require('./GameShipData');
```

File: server/GameBoardData.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+let GameBoardData = class{
+
+    constructor(shipArray){
+        /**
+         * @param {Array[GameShipData]} - Contains all of the ships to the board.
+         */
+        this._guesses = new Array();
+        this._remainingShips = shipArray;
+        this._sunkShip = new Array();
+    }
```
Comment (kvverti) [inline server/GameBoardData.js]: `new Array()` can be replaced with `[]`.

File: server/GameBoardData.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    get ShipCount(){
+        /**
+         * @return {Int} - Returns the number of ships left
+         */
+        return this._remainingShips.length;
+    }
+
+    get LastSunkShipID(){
+        /**
+         * @return {Int} - Returns the id of the last sunk ship. Return null if no ship sunk.
```
Comment (kvverti) [inline server/GameBoardData.js]: Why the capitalization here?

File: server/GameShipData.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
+/**
+ * This class contains the information of a ship on the board.
+ * It will keep track of damages, it's position, and will allow for
+ * guesses on it.
+ */
+
+
+require("./utilities");
+
```
Comment (kvverti) [inline server/GameShipData.js]: Assign to something

File: server/GameShipData.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+         * 
+         * @param {Int} x - The x coordinate of the guess
+         * @param {Int} y - The y coordinate of the guess
+         * @modify - On hit, mark the position as hit.
+         * @return {Guesses} - Returns if it is a hit/miss/skin
+         */
+        for (let i  = 0; i < this._activePositions.length; i++){
+            currentPosition = this._activePositions[i]
+            if (this._activePositions[i].x == x && this._activePositions[i].y == y){
+                this._hitPositions.push(currentPosition);
```
Comment (kvverti) [inline server/GameShipData.js]: Use `===`

File: server/GameShipData.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+        }
+        return Guesses.MISS;
+    }
+
+    isSunk(){
+        /**
+         * @return {Boolean} - Return if the Ship has sunk.
+         */
+        return this._activePositions.length == 0;
+    }
```
Comment (kvverti) [inline server/GameShipData.js]: `===`

File: server/utilities.js [hunk 7]
```diff
@@ -0,0 +1,16 @@
+/**
+ * Contains miscellaneous functions and classes.
+ */
+
+
+let Coordinate = {x: a , y: b};
+
```
Comment (kvverti) [inline server/utilities.js]: This isn't needed. Instead coordinates can be defined inline where they are used

File: client/src/server/GameShipData.js [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+         * Given an x and y position, check to see if the ship is hit.
+         * 
+         * @param {Int} x - The x coordinate of the guess
+         * @param {Int} y - The y coordinate of the guess
+         * @modify - On hit, mark the position as hit.
+         * @return {EnumPosition} - Returns if it is a hit/miss/skin
+         */
+        for (let i  = 0; i < this._activePositions.length; i++){
+            currentPosition = this._activePositions[i]
+            if (this._activePositions[i].x == x && this._activePositions[i].y == y){
```
Comment (kvverti) [inline client/src/server/GameShipData.js]: `const currentPosition`

File: server/BoardData.js [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+const ShipData = require("./ShipData");
+const Guesses = require("./utilities");
+
+let BoardData = class{
+
+    constructor(shipArray){
+        /**
+         * @param {Array[ShipData]} - Contains all of the ships to the board.
+         */
+        this._guesses = [];
```
Comment (kvverti) [inline server/BoardData.js:16]: Looks like this comment was skipped

File: server/GameBoardData.js (hunk 3 above)
Comment (tnguyen57) [reply server/GameBoardData.js]: It was a slight mistake, fixed it now,

**Pull request merged**